### PR TITLE
cmd/skopeo: check errors on close functions for image handling.

### PR DESCRIPTION
**Pull request by erikh**
Co-PR from https://github.com/containers/image/pull/236, more information over there.

### CODE REVIEW

**Comment by erikh** [comment]
Note that this does not even compile yet, it's await the merge from the other patch.

File: cmd/skopeo/inspect.go [hunk 1]
```diff
@@ -49,12 +50,17 @@ var inspectCmd = cli.Command{
...
+	Action: func(c *cli.Context) (retErr error) {
 		img, err := parseImage(c)
 		if err != nil {
 			return err
 		}
-		defer img.Close()
+
+		defer func() {
+			if err := img.Close(); err != nil {
+				retErr = errors.Wrapf(retErr, fmt.Sprintf("could not close image: %v", err))
```
Comment (mtrmac) [inline cmd/skopeo/inspect.go]: I guess this should be consistent with the others WRT parentheses etc.

File: cmd/skopeo/layers.go [hunk 2]
```diff
@@ -36,10 +36,17 @@ var layersCmd = cli.Command{
...
+			if closeErr := rawSource.Close(); closeErr != nil {
+				return errors.Wrapf(err, " (close error: %v)", closeErr)
+			}
+
 			return err
 		}
-		defer src.Close()
+		defer func() {
+			if err := src.Close(); err != nil {
+				retErr = errors.Wrapf(retErr, " (close error: %v)", err)
```
Comment (mtrmac) [inline cmd/skopeo/layers.go]: `retErr` will need to be declared AFAICT

**Comment by erikh** [comment]
updated with the latest containers/image since the close-with-error code was released over there.

**Comment by mtrmac** [comment]
Thanks!

**Pull request merged**